sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2072,6 +2074,386 @@ private Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinkedEntitie
+                string location = response.Headers.OperationLocation;
+
+                var _idToIndexMap = CreateIdToIndexMap(batchInput.Documents);
+
+                return new HealthcareOperation(_serviceRestClient, _clientDiagnostics, location, _idToIndexMap, options.Top, options.Skip, options.IncludeStatistics);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Cancel healthcare prediction job. </summary>
+        /// <param name="operation"> operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        // Look for other alternatives of operation cancellation
+        #pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual async Task<string> StartCancelHealthJobAsync(HealthcareOperation operation, CancellationToken cancellationToken = default)
+        #pragma warning restore AZC0015 // Unexpected client method return type.
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartCancelHealthJobAsync)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsCancelHealthJobHeaders> response = await _serviceRestClient.CancelHealthJobAsync(new Guid(operation.Id), cancellationToken).ConfigureAwait(false);
+
+                return response.Headers.OperationLocation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+
+        }
+
+        /// <summary> Cancel healthcare prediction job. </summary>
+        /// <param name="operation"> Job ID. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        #pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual string StartCancelHealthJob(HealthcareOperation operation, CancellationToken cancellationToken = default)
+        #pragma warning restore AZC0015 // Unexpected client method return type.
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartCancelHealthJobAsync)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsCancelHealthJobHeaders> response = _serviceRestClient.CancelHealthJob(new Guid(operation.Id), cancellationToken);
+
+                return response.Headers.OperationLocation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets collection of healthcare entities from the HealthOperation using async pageable.
```

Comment:
consider updating the docs to say that the user needs to use this method when sending more than 20 documents

Reply:
which reminds me, please mention in all other methods that they will only work if they sent less or equal to 20 documents, otherwise use this method

Reply:
I mean it will work, but all results will not be returned